src/utils/filteration.js

```diff
@@ -0,0 +1,54 @@
+function category_handle(product_data, state) {
+  if (state.category.length !== 0) {
+    return product_data.filter((ele) =>
+      state.category.includes(ele.categoryName)
+    );
+  }
+  return product_data;
+}
+
+function rating_handle(product_data, state) {
```

Comment:
here you can use different file for particular operation so it will look clean and easy to debug